src/compiler/types.ts

```diff
@@ -2719,6 +2719,7 @@ namespace ts {
         inlineSources?: boolean;
         isolatedModules?: boolean;
         jsx?: JsxEmit;
+        jsxFactory?: string;

```

Comment:
It'd probably be cleanest to do this once in checker.ts and then only again if needed in emitter.ts.